scripts/nbconverted/main_Pa_sim_enhance_AtoB.py

```diff
@@ -0,0 +1,283 @@
+             os.path.join(os.path.dirname(os.getcwd()), 'encoded'),
+             os.path.join(os.path.dirname(os.getcwd()), 'models'),
+             os.path.join(os.path.dirname(os.getcwd()), 'output'),
+             os.path.join(os.path.dirname(os.getcwd()), 'stats'),
+             os.path.join(os.path.dirname(os.getcwd()), 'viz')    
+]
+
+# Check if directory exist otherwise create
+for each_dir in base_dirs:
+    analysis_dir = os.path.join(each_dir, analysis_name)
+    
+    if os.path.exists(analysis_dir):
+        print('directory already exists: {}'.format(analysis_dir))
+    else:
+        print('creating new directory: {}'.format(analysis_dir))
+    os.makedirs(analysis_dir, exist_ok=True)
+
+
+# In[13]:
+
+
+# Output the new gene expression values for each sample
+train_input_file = os.path.join(base_dirs[0], analysis_name, "train_model_input.txt.xz")
+data.to_csv(train_input_file, sep='\t', compression='xz', float_format="%.5g")
+
+# Output log file with parameters used to generate simulated data
+log_file = os.path.join(os.path.dirname(os.getcwd()), 'metadata', analysis_name + '_log.txt')
+
+args_dict = {
+    "size of gene set A": geneSetA_size,
+    "size of gene set B": geneSetB_size,
+    "genes in set B": geneSetB,
+    "effect size of genes in set B": effect_size,
+    "threshold of A activation": thresholdA,
+}
+
+with open(log_file, 'w') as f:
+    for key, value in args_dict.items():
+        f.write('%s: %s\n' % (key, value))
+        
+# Output geneA only file using sample gene A selected
+geneA_file = os.path.join(base_dirs[0], analysis_name, geneSetA_pick + ".txt")
+geneA_only.to_csv(geneA_file, sep='\t', float_format="%.5g")
+
+
+# In[14]:
+
+
+get_ipython().run_cell_magic('time', '', '## Train compression methods using simulated data\n\n# Parameters to train nonlinear (VAE) compression method\nlearning_rate = 0.001\nbatch_size = 100\nepochs = 200\nkappa = 0.01\nintermediate_dim = 2775\nlatent_dim = 300\nepsilon_std = 1.0\nnum_PCs = latent_dim\n\nbase_dir = os.path.dirname(os.getcwd())\n\n# Train nonlinear (VAE)\nvae.tybalt_2layer_model(learning_rate, batch_size, epochs, kappa, intermediate_dim,\n                        latent_dim, epsilon_std, base_dir, analysis_name)\n# Train linear (PCA)\npca.pca_model(base_dir, analysis_name, num_PCs)')
+
+
+# In[15]:
+
+
+## Predict gene expression for each sample
+# Prediction based calculation described in interpolate.py
+
+# Define offset vectors in gene space
+data_dir = os.path.join(base_dirs[0], analysis_name)
+target_gene = geneSetA_pick
+percent_low = 5
+percent_high = 95
+
+def_offset.gene_space_offset(data_dir, target_gene, percent_low, percent_high)
+
+model_dir = os.path.join(base_dirs[2], analysis_name)
+encoded_dir = os.path.join(base_dirs[1], analysis_name)
+
+# Define offset vectors for VAE latent space
+def_offset.vae_latent_space_offset(data_dir, model_dir, encoded_dir, latent_dim, target_gene, percent_low,
```

Comment:
what does `percent_low` and `percent_high` do in the VAE and PCA offset calculation?

Reply:
This is how many samples that are used to define the extreme values of gene A that are then used to calculate the offset.  Will add a comment

Reply:
gotcha! Could be possible to define extreme values of gene A as a preprocessing step in `def_offset`. This way it doesn't have to be called for raw, VAE, and PCA. 

Maybe want to think about refactoring at some point if this operation becomes a limiting step.

Reply:
Very good suggestion.  This was discussed in person and I am adding it to list to do.